src/app/app.service.ts

```diff
@@ -25,19 +27,33 @@ export class AppService {
     this.enableVersionListener();
     this.enableCustomCSSListener();
     this.electronService.ipcRenderer.send('appInfo');
-    this.updateError = [
-      ".printer should have required property 'zBabystepGCode'",
-      ".octodash should have required property 'previewProgressCircle'",
-    ];
+
+    // list of all error following an upgrade
+    this.updateError = {
+      ".printer should have required property 'zBabystepGCode'": config => (config.printer.zBabystepGCode = 'M290 Z'),
+      ".plugins should have required property 'tpLinkSmartPlug'": config =>
+        (config.plugins.tpLinkSmartPlug = { enabled: true, smartPlugIP: '127.0.0.1' }),
+      ".octodash should have required property 'previewProgressCircle'": config =>
+        (config.octodash.previewProgressCircle = false),
+      ".octodash should have required property 'turnOnPrinterWhenExitingSleep'": config =>
+        (config.octodash.turnOnPrinterWhenExitingSleep = false),
+    };
   }
 
   // If the errors can be automatically fixed return true here
-  public autoFixError(): boolean {
+  public autoFixErrors(errors: string[]): boolean {
```

Comment:
feel free to call this function migrate, since it is a migrate function now rather than just blindly fixing errors.

Reply:
What about `fixMigrationErrors` ?

Reply:
You made the function, I'm fine with either, so choose the one you like better :)

Reply:
Thanks :)